### PR TITLE
A rough draft of a WFC model

### PR DESCRIPTION
**Pull request by wm333300**
A WFC model that solves sudoku puzzles. In progress...

### CODE REVIEW

File: wavefunctioncollapse.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+def estm_cols(est):
+    ret = {}
+    for elem in range(len(est)):
+        innerlist = []
+        for e in range(len(est[elem])):
+            innerlist.append(est[e][elem])
+        ret[elem] = innerlist
+    return ret
+
+def estm_rc_ids(x, y):
```
Comment (Jandhi) [inline wavefunctioncollapse.py:53]: not clear what rc means, bad naming

File: wavefunctioncollapse.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            innerlist.append(i)
+        outerlist_r.append(innerlist)
+    for elem in range(len(outerlist_r)):
+        innerlist = []
+        for e in range(len(outerlist_r[elem])):
+            innerlist.append(outerlist_r[e][elem])
+        outerlist_c.append(innerlist)
+    return outerlist_r, outerlist_c
+
+def estm_flat(est):
```
Comment (Jandhi) [inline wavefunctioncollapse.py:70]: what is est? again not clear

File: wavefunctioncollapse.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    i = 0
+    l = len(est)
+    ret = {}
+    for elem in range(l):
+        for e in range(len(est[elem])):
+            i += 1
+            ret[i] = est[elem][e]
+    return ret    
+
+board_x = 9
```
Comment (Jandhi) [inline wavefunctioncollapse.py:81]: we should put these variables and functions into a class

File: wavefunctioncollapse.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            
+def entropy(ps):
+    return -1*sum([elem*math.log(elem, 2) for elem in ps])
+
+# entropy tests
+entropy_t = list("1112265477") 
+entropy_d = {elem: entropy_t.count(elem) for elem in entropy_t}
+entropy_p = {elem: entropy_d[elem]/len(entropy_t) for elem in entropy_d.keys()}
+
+def retrieve_blk(id_):
```
Comment (Jandhi) [inline wavefunctioncollapse.py:184]: blk is not worth shortening

**Review by Jandhi — CHANGES_REQUESTED** [review]
I assume this is the code you took online- it may work, but a lot of the variables are named with abbreviations that aren't clear to a fresh reader, and a lot of this should be wrapped into a class or multiple files. I'd try to work your own generalized version before we merge into main, with better naming and organization.

**Comment by wm333300** [comment]
Yeap ill implement these changes. This was more of just putting everything together but I was thinking of breaking it up into files. I'll remove this version

**Pull request closed without merging**